buildSrc/subprojects/build/src/main/groovy/org/gradle/build/ReleasedVersionsFromVersionControl.groovy

```diff
@@ -42,7 +43,7 @@ class ReleasedVersionsFromVersionControl implements ReleasedVersions {
         }.reverse()
         def latestFinalRelease = finalReleases.head()
         def latestNonFinalRelease = [mostRecentSnapshot, mostRecentRc].findAll { it.version > latestFinalRelease.version }.sort { it.buildTimeStamp }.reverse().find()
-        allVersions = ([latestNonFinalRelease] + finalReleases).findAll().findAll { it.version >= lowestInterestingVersion }
+        allVersions = ([latestNonFinalRelease] + finalReleases).findAll().findAll { it.version >= lowestInterestingVersion && it.version.baseVersion < currentBaseVersion}
```

Comment:
Name `allVersions` seems not to be accurate enough. Maybe `allVersionsOfPreviousRelease`?

Reply:
I changed it to `allPreviousVersions`. Sounds good?

Reply:
Absolutely better.